modules/publisher/api.py

```diff
@@ -1,11 +1,11 @@
-import flask
 import modules.authentication as authentication
 import modules.cache as cache
 import os
 from modules.exceptions import (
     ApiResponseDecodeError,
     ApiResponseError,
     ApiResponseErrorList,
+    MacaroonNeedsRefresh
```

Comment:
Minor nitpick on name, I think "requires" is better than "needs".
I'm also wondering if it sounds better as `MacaroonRefreshRequired`. Then it feels namespaced and could do `MacaroonRefreshFailed` and similar things if you needed.

I really like using Exceptions to broadcast the state! :+1:

Reply:
`MacaroonRefreshFailed` will be handled in another PR when I arrive on the refactor of this particular file